dev/tools/gen_localizations.dart

```diff
@@ -107,19 +107,74 @@ String generateString(String s) {
   return output.toString();
 }
 
+/// Simple data class to hold parsed locale. Does not promise validity of any data.
+class Locale {
+  Locale({
+    this.languageCode,
+    this.scriptCode,
+    this.countryCode,
+    this.length,
+    this.origString
+  });
+
+  String languageCode;
+  String scriptCode;
+  String countryCode;
+  int length;          // The number of fields. Ranges from 1-3.
+  String origString;   // Original un-parsed locale string.
+}
+
+/// Simple parser. Expects the locale string to be in the form of 'language_script_COUNTRY'
+/// where the langauge is 2 characters, script is 4 characters with the first uppercase,
+/// and country is 2-3 characters and all uppercase.
+///
+/// 'language_COUNTRY' or 'language_script' are also valid. Missing fields will be null.
+Locale parseLocaleString(String locale) {
```

Comment:
Typically this would be a Locale (LocaleInfo) factory method like `fromString(String locale)`.